src/main.js

```diff
@@ -1,36 +1,31 @@
-const colors = {
-  data: ['magenta', 'cyan', 'firebrick', 'springgreen', 'skyblue'],
-  [Symbol.iterator]() {
-    this.counter = 0;
-    return this; // ваш код
-  },
-  next() {
-    if (this.current === undefined) {
-      this.current = this.counter;
-    }
-    if (this.current < this.data.length) {
-      return {
-        done: false,
-        value: this.data[this.current++],
-      };
-    } else {
-      delete this.current;
-      this.current = 0;
-      return {
-        done: false,
-        value: this.data[this.current++],
-      };
-    }
-  },
-};
+import {dataHomes} from "./data/data.js";
 
-function changeColor(colors) {
-  return function (event) {
-    event.target.style.backgroundColor = colors.next().value;
-  };
+const homesCards = document.querySelector('.homes-cards');
```

Comment:
I reccomend in the future to use for js id or special classes with mark
for example `homes-cards--js` or something like this, because it can be **protection** for deleting during style changing